assets/source/setup-guide/app/components/SaveSettingsButton.js

```diff
@@ -12,13 +13,33 @@ import {
 	useSettingsDispatch,
 	useCreateNotice,
 } from '../helpers/effects';
+import getContextByPath from '../helpers/get-context-by-path';
 
+/**
+ * Clicking on "… Save changes" button.
+ *
+ * @event wcadmin_pfw_save_changes_button_click
+ */
+
+/**
+ * Save Settings button component
+ *
+ * @fires wcadmin_pfw_save_changes_button_click
+ *
+ * @return {JSX.Element} Rendered element
+ */
 const SaveSettingsButton = () => {
 	const isSaving = useSettingsSelect( 'isSettingsUpdating' );
+	const updatedData = useSettingsSelect( 'getUpdatedData' );
 	const setAppSettings = useSettingsDispatch( true );
 	const createNotice = useCreateNotice();
 
 	const saveSettings = async () => {
+		recordEvent( 'pfw_save_changes_button_click', {
+			updatedData,
```

Comment:
🚧  
I think the naming and value meaning of `updatedData` are confusing.

From the naming perspective, the event itself tracks the attempt of clicking the save button, but it doesn't matter if the settings are successfully saved or not. The **updated** in naming is easily interpreted to mean that the data update is complete/successful.

From the value meaning perspective, no matter each property appears in the `updatedData` or its value is `true`/`false`, they don't have a stable meaning. For example, if `updatedData` has

- `enable_debug_logging` property, it still couldn't tell where the `enable_debug_logging` is actually what value, because the event is called regardless of the API request is successful or not.
- `enable_debug_logging` property with `true` or `false` value, even if we ignore whether the data is successfully saved, it doesn't mean that the setting is "updated" to `true`/`false` at that moment.
    If a user checks and then unchecks the same setting, that setting will appear in `updatedData` with the same value against the corresponding value in DB. This means it not always could be interpreted to mean "a setting is updated to true/false", it could be no change.

    https://user-images.githubusercontent.com/17420811/146323294-793b8140-6f04-4664-9642-0f847933fafc.mp4

    (The `false` in the API response `pinterest_for_woocommerce: false` means that no update occurred in DB.)

I would suggest reconsidering, suppose as a data analyst, whether there is a known scenario that is analyzed by the clicking event of the "Save" button and group the settings data to be updated (but not necessarily actually be updated successfully) at that moment?

- If no, would it be proper to remove the `updatedData` at all?
- If yes, it'd be better to make the naming and the value meaning stable and clear.

Reply:
> If yes, it'd be better to make the naming and the value meaning stable and clear.

What about, `dataToBeSaved`? Any suggestion for naming?

> I would suggest reconsidering, suppose as a data analyst, whether there is a known scenario that is analyzed by the clicking event of the "Save" button and group the settings data to be updated (but not necessarily actually be updated successfully) at that moment?

Thats a valid point, some thoughts: 

Could be considered as more accurate to register the property even if the user checks and then unchecks again the property. Which is more close to the real user action. The user attempted to save changes and manipulated that props and as analyst I want to know that.

On the other side, also could be possible to compare the `updatedData` values against the previous values and only update them if there is a change. 

> f no, would it be proper to remove the updatedData at all?

By removing  `updatedData` it would be needed to trigger the event based on save instead of click. Which as far as I now is not what we want to achieve.  

Opinions, thoughts?

Reply:
> If yes, it'd be better to make the naming and the value meaning stable and clear.

IN regards this... Checking the code again, the current action events were called `updateSettings` and also there is other one named `isUpdating`  so, I took as granted that `update` is the appropriate verb.

Reply:
> Could be considered as more accurate to register the property even if the user checks and then unchecks again the property. Which is more close to the real user action. The user attempted to save changes and manipulated that props and as analyst I want to know that.

This could be a possible scenario, I'm a bit confused about what insights could be figured out from it though. because it's only possible to know that the user has manipulated this option on UI. It might not possible to tell what the user intended, for example: 
- they forgot they had changed the option and then changed it back
- or they know they changed the option, but then changed their mind and restore the option back
- or they did want to and did change it (and have to find a way to compare it from recorded analysis data to the actual saved option value) 

> By removing `updatedData` it would be needed to trigger the event based on save instead of click. Which as far as I now is not what we want to achieve.

Whether you decide *yes* or *no*, the reason I suggested reconsidering was, even though the spreadsheet requirements list "indication of what is being saved/changed", you could still suspect/question whether it might have wrong assumptions and then correct it if so.

I was mainly concerned if the data recorded does not have a concrete and stable meaning, then it may not work for the intended scenario, or may further affect the analysis results and mislead to the wrong insights. So if you decide to send this tracking event with **settings** as event properties, it would be good to also use JSDoc, such as [wcadmin_pfw_get_started_faq](https://github.com/woocommerce/pinterest-for-woocommerce/blob/429af73ca403453f802fed7d34e43d44b784dac4/assets/source/setup-guide/app/views/LandingPageApp.js#L211-L214), to provide the concrete meaning for those event properties. With the well-documented definition, we can easily [refer to them here](https://github.com/woocommerce/pinterest-for-woocommerce/blob/429af73ca403453f802fed7d34e43d44b784dac4/TRACKING.md) without having to deduce their meaning from code or semantically inexpressible naming in the future.

Reply:
State issues fixed in https://github.com/woocommerce/pinterest-for-woocommerce/pull/295/commits/3babdc6464b1d91839a42c8aa4e46c5f160bf393

I removed completely the implemented `updateState`

Reasons: 

- Reducing complexity
- @eason9487 Feedback
- Checking a similar implementation in `Analytics -> Settings -> Save` I saw that it just tracks all the settings, no matter if the user changed or not. 
- Allows to format the data as `Analytics -> Settings -> Save` with separated properties + the context
- Only allow pretended data to be tracked, excluding other sensitive data like `account_info` or trying `track_tag`

Now the Event is tracked like this:

![Screenshot 2021-12-16 at 13 06 46](https://user-images.githubusercontent.com/5908855/146376717-c507ffb8-f10e-43cb-8129-4ea447d85b90.png)

💅 

Inside `assets/source/setup-guide/app/helpers/prepare-for-tracking.js` I set `TRACKABLE_SETTINGS` constant for allowing properties in settings to be tracked. 

This is because I consider more critical to prevent sensitive data to be tracked than the tracking itself. In other words,I think is better to don't track a new property we forgot to add inside TRACKABLE_SETTINGS rather than forgot to block sensitive data on each setting ting addition. 

💅 

Notice a better and cleaner solution for this would be to add the settings properties (trackable) inside `state.settings` and  Connection properties (not trackable) outside `state.settings` for example `state.connection`.  But that would require more refactoring and it's out scope of this PR.

I refer again to #302 for going further on this

Reply:
> , it would be good to also use JSDoc, such as [wcadmin_pfw_get_started_faq](https://github.com/woocommerce/pinterest-for-woocommerce/blob/429af73ca403453f802fed7d34e43d44b784dac4/assets/source/setup-guide/app/views/LandingPageApp.js#L211-L214), to provide the concrete meaning for those event properties.

Adding missing props here https://github.com/woocommerce/pinterest-for-woocommerce/pull/295/commits/5d4eaa9bd9cb593e3496939ca06a551d28714f2f